src/query.rs

```diff
@@ -713,9 +714,15 @@ mod tests {
         indexed_crate: &VersionedIndex<'_>,
         crate_pair_name: &String,
         crate_version: &str,
+        indexed_query: Arc<IndexedQuery>,
```

Comment:
I would move this arg next to `semver_query` both here and in the other function, because it's directly tied to `semver_query` in a non-obvious way. If they get out of sync, we'll have a bad time.

I'd also leave a comment explaining the connection between them.